### DISCUSSION TITLE
How to build the C# compiler after forking roslyn

### DISCUSSION THREAD

**Discussion by The-Futurist** [dotnet/roslyn#57302] [Q&A] [answered]
Are there any docs or guidelines about how to build and run a copy of the C# 10.0 compiler? I'd like to get to a point where I am able to make experimental changes to the implementation as part of some experiments into the language.

Do I need a bunch of tools beside Visual Studio? 

Any info appreciated!

**Comment by alrz** [comment, marked as answer]
https://github.com/dotnet/roslyn/blob/main/docs/contributing/Building%2C%20Debugging%2C%20and%20Testing%20on%20Windows.md

**Reply by CyrusNajmabadi** [reply]
> Do I need a bunch of tools beside Visual Studio?

You do not need Visual Studio.  The compiler can be built just using the SDK.  (if this was not possible then you couldn't build it on Linux for example).
